railties/test/generators/shared_generator_tests.rb

```diff
@@ -344,14 +344,14 @@ def test_dev_option
   end
 
   def test_edge_option
-    Rails.stub(:gem_version, Gem::Version.new("2.1.0")) do
+    Rails.stub(:gem_version, Gem::Version.new("7.1.0")) do
```

Comment:
This one feels bad as well, it would be great if we didn't have to update it in the future

Reply:
Also looking at this one, we could do something like read the `RAILS_VERSION` file, but there are separate tests for alpha and non-alpha, and that file isn't the same on stable branches. So we could craft these strings ourselves, but that seems like overkill for two tiny test cases.

Reply:
The whole point of using an unambiguously ancient version here is that it should never need to change.

Reply:
For `test_edge_option` the test generates the following output 9 times, but _only_ when running the `app_generator_test` and not `plugin_generator_test` even though they both include it.

```
warning: rejected HEAD because shallow roots are not allowed to be updated
Could not find gem 'rails' in https://github.com/rails/rails.git (at 2-1-stable@b5d759f).
error: Could not read 52b85b91abbf45a4af37d9383d44bf25a0699a75
```

I will continue to investigate.

Reply:
Yeah -- it doesn't surprise me that actual `2-1-stable` makes bundler unhappy, but this test has no reason to need to be talking to the network / seeing the actual repo contents.